crepecake/src/main/java/com/google/cloud/tools/crepecake/image/CachedLayer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.blob.BlobDescriptor;
+import com.google.cloud.tools.crepecake.blob.BlobStream;
+import com.google.cloud.tools.crepecake.blob.BlobStreams;
+import java.io.File;
+
+/**
+ * A {@link Layer} that has been written out (i.e. to a cache) and has its file-backed content BLOB,
+ * digest, size, and diff ID.
+ */
+public class CachedLayer extends Layer {
+
+  private final File file;
```

Comment:
How are these related? any file with any descriptor and any digest? or do they have to match? If they have to match somehow, will we detect it here if they don't?

Reply:
When caching, the UnwrittenLayer is written out as its BLOB, descriptor, and diffId. There will be no checks at runtime in CachedLayer for matching and the user of CachedLayer must ensure that the descriptor and diffId matches the BLOB stored in the file. I will add a note about this in the javadoc.

Reply:
Could the constructor be non-public to make it harder to create a layer where these pieces don't match? Or could we reorganize the API so they always match?